src/coreApi/registerFunction.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { FunctionCallback, FunctionMetadata, RpcBindingInfo } from '@azure/functions-core';
+import { v4 as uuid } from 'uuid';
+import { AzureFunctionsRpcMessages as rpc } from '../../azure-functions-language-worker-protobuf/src/rpc';
+import { Disposable } from '../Disposable';
+import { InternalException } from '../utils/InternalException';
+import { WorkerChannel } from '../WorkerChannel';
+
+export function registerFunction(
+    channel: WorkerChannel,
+    metadata: FunctionMetadata,
+    callback: FunctionCallback
+): Disposable {
+    if (channel.hasIndexedFunctions) {
+        throw new InternalException('A function can only be registered during app startup.');
+    }
+    const functionId = uuid();
+
+    const rpcMetadata: rpc.IRpcFunctionMetadata = metadata;
+    rpcMetadata.functionId = functionId;
+    // `rawBindings` is what's actually used by the host
+    // `bindings` is used by the js library in both the old host indexing and the new worker indexing
+    rpcMetadata.rawBindings = Object.entries(metadata.bindings).map(([name, binding]) => {
```

Comment:
Due to our lint/formatting rules, these are our two options (because the line is apparently too long):

```
rpcMetadata.rawBindings = Object.entries(metadata.bindings).map(([name, binding]) => {
    return convertToRawBinding(name, binding);
});
```

```
rpcMetadata.rawBindings = Object.entries(metadata.bindings).map(([name, binding]) =>
    convertToRawBinding(name, binding);
);
```

Out of those two, I prefer the first